### PR TITLE
FBE-1502: email unsubscribe webhook

### PR DESCRIPTION
**Pull request by dhull**

### CODE REVIEW

File: source/includes/_email_recipient_auth_callback.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+## Response Payload
+
+> Example Response
+
+```json
+["allowed-friend@example.com"]
+```
+
+Friendbuy expects the response to the email recipient authorization callback to be a list of the allowed email addresses. The share emails for recipients in the returned list of allowed email addresses will be sent, and the share emails to recipients not in the returned list will be blocked.
```
Comment (britainjs) [inline source/includes/_email_recipient_auth_callback.md:40]: Do we actually block the email (i.e. do we prevent that email from being sent to in the future)? If not, we may want to say something like "Emails not in the returned list will not receive a share email" or something like that.

File: source/includes/_webhooks.md [hunk 2]
```diff
@@ -156,3 +156,44 @@ The `incentive` object in the payload has the following structure:
...
 | currency   | string | The currency of the incentive \(i.e. "USD"\).                              |
+
+## Email Unsubscribe Webhook
+
+> Example Payload
+
+``` json
+{
+  "id": "fd2f5766-8c27-4bfe-b3d3-9f41e186b13e",
+  "type": "emailOptOut",
```
Comment (britainjs) [inline source/includes/_webhooks.md:167]: We should probably be consistent between the type and the labeling we use. So both should be Email Unsubscribe or both should be Email Opt Out.

**Review by britainjs — COMMENTED** [review]
Couple of minor notes on copy, but looks good overall.

File: source/includes/_email_recipient_auth_callback.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+### Retry Behavior
+
+If the Email Recipient Authorization Callback fails or if it receives a failure response, the sending of those share emails will be temporarily blocked and the callback will be retried later, on Friendbuy's email retry schedule.  If, after six retries the callback is still failing, the emails will not be sent.
```
Comment (nimishfb) [inline source/includes/_email_recipient_auth_callback.md:44]: We should clarify what Friendbuy's email retry schedule is or link to it if its documented

File: source/includes/_email_recipient_auth_callback.md (hunk 1 above)
Comment (dhull) [reply source/includes/_email_recipient_auth_callback.md:40]: The email remains in the Message_EmailQueue table but its status is set to "unauthorized" which prevents it from ever being sent. I'm happy to change the wording though.

Comment (britainjs) [reply source/includes/_email_recipient_auth_callback.md:40]: Yeah, we should make it clear that it's not a permanent block. I can see people getting confused about that.

File: source/includes/_email_recipient_auth_callback.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+| type      | "emailAuth"      | Indicates that this is an Email Recipient Authorization callback. |
+| data      | array of strings | An array of recipient email addresses.                            |
+| createdOn | ISO timestamp    | The time the request was sent.                                    |
+
+## Response Payload
+
+> Example Response
+
+```json
+["allowed-friend@example.com"]
```
Comment (nimishfb) [inline source/includes/_email_recipient_auth_callback.md:37]: This should be allowed.friend@example.com

File: source/includes/_email_recipient_auth_callback.md (hunk 1 above)
Comment (dhull) [reply source/includes/_email_recipient_auth_callback.md:40]: I have changed the text to distinguish more clearly between "email message" and "email address" and avoided the word "block".

File: source/includes/_email_recipient_auth_callback.md (hunk 3 above)
Comment (dhull) [reply source/includes/_email_recipient_auth_callback.md:44]: Done.

File: source/includes/_email_recipient_auth_callback.md (hunk 4 above)
Comment (dhull) [reply source/includes/_email_recipient_auth_callback.md:37]: Thanks for your (and Adam's) eagle eyes.

File: source/includes/_webhooks.md (hunk 2 above)
Comment (dhull) [reply source/includes/_webhooks.md:167]: After a discussion in Slack, my understanding of the consensus was that it's OK to keep both "unsubscribe" and "opt-out".

Comment (nimishfb) [reply source/includes/_webhooks.md:167]: @dhull using two terms for one thing is extremely confusing. Let's please use Opt-out if that's what we've been using everywhere else.

**Review by nimishfb — COMMENTED** [review]
LGTM

File: source/includes/_webhooks.md (hunk 2 above)
Comment (dhull) [reply source/includes/_webhooks.md]: Now changed to "Email Opt-Out"/emailOptOut.

**Review by britainjs — APPROVED** [review]
LGTM

**Pull request merged**